### PR TITLE
fix: enable pinning of typescript via "resolutions"

### PR DESCRIPTION
**Pull request by levino**

### CODE REVIEW

**Comment by levino** [comment]
I am not sure if `yarn` would have suffered from the same problems when `typescript` were no root dependency. Might or might not be the case.

File: package.json [hunk 1]
```diff
@@ -34,9 +34,11 @@
...
     "parcel": "2.8.0",
     "prettier": "^2.7.1",
-    "turbo": "^1.6.3"
+    "turbo": "^1.6.3",
+    "typescript": "^4"
   },
   "resolutions": {
-    "typescript": "~4.7"
+    "typescript": "~4.7",
+    "typescript:comment": "https://github.com/parcel-bundler/parcel/issues/8419"
```
[the remark this answers is not in the document]
Comment (shubham-kaushal) [reply package.json:42]: Adding this is creating the error `ERR_PNPM_INVALID_OVERRIDE_SELECTOR`

**Comment by shubham-kaushal** [comment]
> I am not sure if `yarn` would have suffered from the same problems when `typescript` were no root dependency. Might or might not be the case.

I'll try to check with yarn as well to confirm.

File: package.json [hunk 2]
```diff
@@ -34,9 +34,11 @@
     "node-abort-controller": "3.0.1",
     "parcel": "2.8.0",
     "prettier": "^2.7.1",
-    "turbo": "^1.6.3"
+    "turbo": "^1.6.3",
+    "typescript": "^4"
```
[the remark this answers is not in the document]
Comment (shubham-kaushal) [reply package.json:38]: noted

**Comment by shubham-kaushal** [comment]
image
@levino Thanks for your help!🙏 Builds are working fine now. But looks like Tree Shaking and Code Splitting is not working properly. The bundle size is big. I am releasing a dev version now to cross-check, will update you.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (shubham-kaushal) [reply package.json:42]: error 
image


and comment is not allowed in json

**Comment by shubham-kaushal** [comment]
The bundle size with the parcel, which should reduce further to 12kb minified, currently it's 36kb minified
https://bundlephobia.com/package/tastycss@0.0.0-dev-20221117103514

**Comment by levino** [comment]
I have no idea why you wanted to use parcel in the first place. I am just a random passer by who had https://github.com/parcel-bundler/parcel/issues/8419#issuecomment-1318254018 and found a workaround after @TechQuery and @mischnic did not like my fix. Since you were unable to use the workaround to get parcel running, I helped you. I cannot say if the bundle size is small or large. That is another question / issue you need to raise with parcel.

**Comment by shubham-kaushal** [comment]
Sorry, I thought you are one of the team members of the parcel team.

I was using `tsup` which is using `esbuild` under the hood and was facing tree-shaking issues in our both TastyCSS and [Jenga UI](https://github.com/OutpostHQ/jengaui) library, especially with Jenga UI, more. tsup is good but it lacks community and support, I tried everything recommended in tsup doc https://tsup.egoist.dev/#tree-shaking but it didn't solve the issue of tree shaking in Jenga UI.

So, thought of trying parcel after looking at this comparison https://github.com/devongovett/parcel-vs-vite-vs-turbopack-hmr and doc https://parceljs.org/features/production/

Thanks again for your help! 🙏

**Pull request merged**